vignettes/adrs_pcwg3.Rmd

```diff
@@ -0,0 +1,516 @@
+
+      # Scenario 6: Soft Tissue = NED + Bone Lesion = NE -> Overall response = NE
+      AVALC.SFTSRESP == "NED" & AVALC.BONERESP == "NE" ~ "NE",
+
+      # Scenario 7: Soft Tissue = SD + Bone Lesion = Non-PD, PDu, NED, or NE -> Overall response = SD
+      AVALC.SFTSRESP == "SD" & AVALC.BONERESP %in% c("Non-PD", "PDu", "NED", "NE") ~ "SD",
+
+      # Scenario 8: Soft Tissue = Non-CR/Non-PD + Bone Lesion = Non-PD, PDu, NED, or NE -> Overall response = Non-CR/Non-PD
+      AVALC.SFTSRESP == "Non-CR/Non-PD" & AVALC.BONERESP %in% c("Non-PD", "PDu", "NED", "NE") ~ "Non-CR/Non-PD",
+
+      # Scenario 9: Soft Tissue = PR + Bone Lesion = Non-PD, PDu, NED, or NE -> Overall response = PR
+      AVALC.SFTSRESP == "PR" & AVALC.BONERESP %in% c("Non-PD", "PDu", "NED", "NE") ~ "PR",
+
+      # Scenario 10: Soft Tissue = CR + Bone Lesion = Non-PD, PDu, NE -> Overall response = PR (default to PR without screening info)
+      AVALC.SFTSRESP == "CR" & AVALC.BONERESP %in% c("Non-PD", "PDu", "NE") ~ "PR",
+
+      # Scenario 11: Soft Tissue = CR + Bone Lesion = NED -> Overall response = CR
+      AVALC.SFTSRESP == "CR" & AVALC.BONERESP == "NED" ~ "CR",
+
+      # Default: If conditions are not met, assign NA
+      TRUE ~ NA_character_
+    ),
+    PARAMCD = "OVRLRESC",
+    PARAM = "Overall Tumor Response by Investigator - Derived",
+    PARAMN = 4,
+    PARCAT1 = "PCWG3 and RECIST 1.1"
+  )
+)
+``` 
+
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adrs,
+   display_vars = exprs(USUBJID,PARAM,PARAMCD,PARCAT1,AVALC,AVISIT,ADT)
+)
+```
+
+
+## Derive `AVAL` (Numeric tumor response from AVALC values)
+
+```{r}
+adrs <- adrs %>%
+  mutate(
+    AVAL = case_when(
+      AVALC == "CR" ~ 1,            # Complete Response
+      AVALC == "PR" ~ 2,            # Partial Response
+      AVALC == "SD" ~ 3,            # Stable Disease
+      AVALC == "NED" ~ 4,           # No Evidence of Disease
+      AVALC == "Non-CR/Non-PD" ~ 5, # Neither Complete Response nor Progressive Disease
+      AVALC == "NE" ~ 6,            # Not Evaluable
+      AVALC == "PDu" ~ 7,           # Progressive Disease Unconfirmed
+      AVALC == "PD" ~ 8,            # Progressive Disease
+      TRUE ~ NA_real_               # Default for unexpected/missing AVALC values
+    )
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  adrs %>%
+    arrange(!!!get_admiral_option("subject_keys"), PARAMN,),
+  display_vars = exprs(USUBJID,PARCAT1,PARAMCD, PARAM,PARAMN,AVISIT,AVISITN,ADT,AVALC,AVAL)
+)
+```
+
+
+
+### Best Overall Response (BOR) and Confirmed Best Overall Response (CBOR)
+
+BOR represents the Best Overall Responses observed during the study, reflecting valid tumor responses such as Complete Response (CR), Partial Response (PR), Stable Disease (SD), and Progressive Disease (PD). *Progressive Disease Unconfirmed (PDu)* is not included in BOR unless it is confirmed as PD based on the 2+2 flare rule, as per PCWG3 and RECIST 1.1 guidelines.
```

Comment:
I think we should change to
```suggestion
BOR represents the Best Overall Responses observed during the study, reflecting valid tumor responses such as Complete Response (CR), Partial Response (PR), Stable Disease (SD), and Progressive Disease (PD). For BOR and CBOR we consider *Progressive Disease Unconfirmed (PDu)* as PD if it is confirmed as PD based on the 2+2 flare rule, as per PCWG3 and RECIST 1.1 guidelines or if it is the last assessment.
```
according to what we've discussed in the last meeting.